File: www/src/lib/models/activity.ts
import type { User } from './user';

export interface Activity {
    createdTime: string;
    id: string;
    kind: ActivityKind;
    actorId: string;
    actor: User;
    projectId: string | null;
    taskId: string | null;
    metadata: string | null;
}

export enum ActivityKind {
    Created = 'created',

    Commented = 'commented',
    Assigned = 'assigned',
    Unassigned = 'unassigned',
    StatusChanged = 'status_changed',
    PriorityChanged = 'priority_changed',
    TitleChanged = 'title_changed',
}

File: api/WebApp.Domain/Entities/Activity.cs
using NodaTime;

namespace WebApp.Domain.Entities;

public sealed record Activity
{
    public Instant CreatedTime { get; init; }
    public ActivityId Id { get; init; }
    public UserId? ActorId { get; init; }
    public User? Actor { get; init; }
    public ProjectId? ProjectId { get; init; }
    public Project? Project { get; init; }
    public TaskId? TaskId { get; init; }
    public TaskEntity? Task { get; init; }
    public ActivityKind Kind { get; init; }
    public string? Metadata { get; init; }
}
